File: SDKs/googleapis.com/androidpublisher/v3/java/lib/src/main/java/openapisdk/models/shared/RegionalBasePlanConfig.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * RegionalBasePlanConfig
 * Configuration for a base plan specific to a region.
**/
public class RegionalBasePlanConfig {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("newSubscriberAvailability")
    public Boolean newSubscriberAvailability;
    public RegionalBasePlanConfig withNewSubscriberAvailability(Boolean newSubscriberAvailability) {
        this.newSubscriberAvailability = newSubscriberAvailability;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("price")
    public Money price;
    public RegionalBasePlanConfig withPrice(Money price) {
        this.price = price;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("regionCode")
    public String regionCode;
    public RegionalBasePlanConfig withRegionCode(String regionCode) {
        this.regionCode = regionCode;
        return this;
    }
}

File: SDKs/googleapis.com/androidpublisher/v3/python/src/sdk/models/shared/regionalbaseplanconfig.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import money as shared_money


@dataclass_json
@dataclasses.dataclass
class RegionalBasePlanConfig:
    r"""RegionalBasePlanConfig
    Configuration for a base plan specific to a region.
    """
    
    new_subscriber_availability: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('newSubscriberAvailability') }})
    price: Optional[shared_money.Money] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('price') }})
    region_code: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('regionCode') }})
    
